hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/SCMContainerPlacementRackAware.java

```diff
@@ -386,6 +386,7 @@ protected int getMaxReplicasPerRack(int numReplicas, int numberOfRacks) {
 
   @Override
   protected int getRequiredRackCount(int numReplicas) {
-    return REQUIRED_RACKS;
+    int racks = networkTopology != null ? networkTopology.getRackCount() : 1;
+    return Math.min(REQUIRED_RACKS, racks);
```

Comment:
From my understanding, this is the only change which fixes the particular issue. I think it would be better to create another refactoring jira, if you want to change the other interfaces.